ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -331,6 +443,7 @@ private module Cached {
       p instanceof SplatParameter
     } or
     TSelfParameterNode(MethodBase m) or
+    TLambdaSelfParameterNode(Callable c) { lambdaCreationExpr(_, _, c) } or
```

Comment:
I think we should rename this to `TLambdaSelfReferenceNode`. As it stands it sounds too much like it's related to the `self` parameter somehow, which it isn't.

Blocks can actually receive a `self` argument that overrides its captured view of `self`. When and if we add support for that, we'll need an actual `self` parameter for lambdas, and then this name will just get even more confusing.

Reply:
You mean for blocks passed to [`module_eval`](https://apidock.com/ruby/Module/module_eval) and similar methods?

Reply:
@aibaars Yes, `module_eval`, `class_eval`, `instance_eval`, or any method that that passes its block to one of those methods, such as `RSpec.describe` in a test or `rescue_from` in an action controller.